llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -3425,6 +3445,11 @@ bool SIInstrInfo::isFoldableCopy(const MachineInstr &MI) {
   case AMDGPU::V_ACCVGPR_MOV_B32:
   case AMDGPU::AV_MOV_B32_IMM_PSEUDO:
     return true;
+  case AMDGPU::AV_MOV_B64_IMM_PSEUDO:
+    // TODO: We could fold this, but it's a strange case. The immediate value
+    // can't be directly folded into any real use. We would have to spread new
+    // immediate legality checks around and only accept subregister extracts for
+    // profitability.
```

Comment:
Sorry, forgive my ignorance but why is this? Is the 2x inline immediate constraint the bottleneck for now?

Reply:
Because no real instruction has the same format of the use. It shouldn't matter, since we only introduce these later. It's not actually relevant